ludwig/utils/audio_utils.py

```diff
@@ -0,0 +1,110 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import numpy as np
+from scipy.signal.windows import get_window
+from scipy.signal import lfilter
+
+def _pre_emphasize_data(data, emphasize_value=0.97):
+    filter_window = np.asarray([1, -emphasize_value])
+    pre_emphasized_data = lfilter(filter_window, 1, data)
+    return pre_emphasized_data
+
+def get_length_in_samp(sampling_rate_in_hz, length_in_s):
+    return int(sampling_rate_in_hz * length_in_s)
+
+def get_group_delay(raw_data, sampling_rate_in_hz, window_length_in_s, window_shift_in_s, num_fft_points, window_type):
+    X_stft_transform = _get_stft(raw_data, sampling_rate_in_hz, window_length_in_s, window_shift_in_s, num_fft_points, window_type=window_type)
+    Y_stft_transform = _get_stft(raw_data, sampling_rate_in_hz, window_length_in_s, window_shift_in_s, num_fft_points, window_type=window_type, data_transformation='group_delay')
+    X_stft_transform_real = np.real(X_stft_transform)
+    X_stft_transform_imag = np.imag(X_stft_transform)
+    Y_stft_transform_real = np.real(Y_stft_transform)
+    Y_stft_transform_imag = np.imag(Y_stft_transform)
+    nominator = np.multiply(X_stft_transform_real, Y_stft_transform_real) + np.multiply(X_stft_transform_imag, Y_stft_transform_imag)
+    denominator = np.square(np.abs(X_stft_transform))
```

Comment:
X_stft_transform should be the real part already, so np.abs is redundant here.

Reply:
naming was probable confusing: X_stft_tranform stood for the 'normal' stft transformed part in the group_delay feature where as Y_stft_transform stood for the 'stft transformed' of the signal multiplied by a ramping function: 03992c1